contracts/VeListaAutoCompounder.sol

```diff
@@ -0,0 +1,270 @@
+        uint256 feeAmount = (_rewardAmount * feeRate) / 10000;
+        uint256 price = oracle.peek(address(lista));
+        uint256 feeValue = (feeAmount * price) / 1e8;
+
+        if (feeValue >= maxFee) {
+            _amtToCompound = _rewardAmount - (maxFee * 1e8) / price;
+        } else {
+            _amtToCompound = _rewardAmount - feeAmount;
+        }
+    }
+
+    /**
+     * @dev Update the contract variables
+     * @param what The variable to update; bytes32 encoded of the state variable name
+     * @param data The new value
+     */
+    function file(
+        bytes32 what,
+        uint256 data
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (what == "feeRate") {
+            require(data <= maxFeeRate, "Invalid feeRate");
+            feeRate = data;
+        } else if (what == "maxFeeRate") {
+            require(feeRate <= data && data <= 10000, "Invalid maxFeeRate");
+            maxFeeRate = data;
+        } else if (what == "maxFee") {
+            require(
+                (feeRate * autoCompoundThreshold) / 10000 <= data,
+                "Invalid maxFee"
+            );
+            maxFee = data;
+        } else if (what == "autoCompoundThreshold") {
+            require(
+                (feeRate * data) / 10000 <= maxFee,
+                "Invalid autoCompoundThreshold"
+            );
+            autoCompoundThreshold = data;
+        } else {
+            revert("Unrecognized variable");
+        }
+    }
+
+    /**
+     * @dev Update the stored addresses
+     * @param what The contract address to update; bytes32 encoded of the contract(state variable) name
+     * @param data The new value
+     */
+    function file(
+        bytes32 what,
+        address data
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(data != address(0), "Zero address provided");
+        if (what == "lista") {
+            require(data != address(lista), "lista already set");
+            require(data == address(veLista.token()), "Invalid Lista token");
+            lista = IERC20(data);
+        } else if (what == "veLista") {
+            require(data != address(veLista), "veLista already set");
+            require(
+                data == address(veListaDistributor.veLista()),
+                "Invalid veLista"
+            );
+            veLista = IVeLista(data);
+        } else if (what == "veListaDistributor") {
+            require(
+                data != address(veListaDistributor),
+                "veListaDistributor already set"
+            );
+            veListaDistributor = IVeListaDistributor(data);
+            emit VeListaDistributorUpdated(data);
+        } else if (what == "oracle") {
+            require(data != address(oracle), "oracle already set");
+            oracle = OracleInterface(data);
+            emit OracleUpdated(data);
+        } else if (what == "feeReceiver") {
+            require(data != feeReceiver, "feeReceiver already set");
+            feeReceiver = data;
```

Comment:
add an event here

Reply:
done in 13945e308d471b9045e67d3d07c56caa7e1c1455